### PR TITLE
Feature#14 dashboard controller and styles

### PR DESCRIPTION
**Pull request by ever864**
# Description
What did you implemented/Why did you implemented this:
 - Fix importmap and setup JS files.
 - Add pages dashboard and it's styles.
 - Add routes for pages dashboard.
 - Fix blank line in the end of files.
 - Add rubocop-rails gem.
 - Use renders for sections in index dashboard. 
- Create method in helper for svg icons.

 Examples:
 - What?: Pages in dashboard have the same nav and added graphics to dashboard root page.
 - Why?: The user needs to see a nav (menu) so that can visit any page from dashboard. 
 
## Screenshots
Desktop
image
Tablet
image
Mobile
image

### CODE REVIEW

**Review by acorzo — APPROVED** [review]
Good Job 👍

**Review by AgustinPalmaM — APPROVED** [review]
lgtm

File: app/views/dashboard/inventory.html.erb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<body class="h-screen bg-gray-100 font-sans">
+    
+   <%= render 'nav' %>
+
+   <div class="flex flex-row flex-wrap flex-1 flex-grow content-start pl-16">
+      <div id="dash-content" class="bg-gray-200 py-6 lg:py-0 w-full lg:max-w-sm flex flex-wrap content-start">
+        <h1 class="pl-2">Inventory</h1>
+      </div>
+   </div>
+</body>
```
Comment (JuanequeX) [inline app/views/dashboard/inventory.html.erb:10]: Same here with the identation

File: app/views/dashboard/providers.html.erb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<body class="h-screen bg-gray-100 font-sans">
+
+   <%= render 'nav' %>
+
+   <div class="flex flex-row flex-wrap flex-1 flex-grow content-start pl-16">
+      <div id="dash-content" class="bg-gray-200 py-6 lg:py-0 w-full lg:max-w-sm flex flex-wrap content-start">
+        <h1 class="pl-2">providers</h1>
+      </div>
+   </div>
+</body>
```
Comment (JuanequeX) [inline app/views/dashboard/providers.html.erb:10]: Same here with the identation

File: app/views/dashboard/sales_and_purcharses.html.erb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<body class="h-screen bg-gray-100 font-sans">
+   <%= render 'nav' %>
+   <div class="flex flex-row flex-wrap flex-1 flex-grow content-start pl-16">
+      <div id="dash-content" class="bg-gray-200 py-6 lg:py-0 w-full lg:max-w-sm flex flex-wrap content-start">
+        <h1 class="pl-2">ventas y compras</h1>
+      </div>
+   </div>
```
Comment (JuanequeX) [inline app/views/dashboard/sales_and_purcharses.html.erb:7]: Same here with the identation

**Review by JuanequeX — CHANGES_REQUESTED** [review]

File: app/helpers/dashboard_helper.rb [hunk 4]
```diff
@@ -0,0 +1,2 @@
+module DashboardHelper
+end
```
Comment (YilKanda) [inline app/helpers/dashboard_helper.rb:2]: If you don't occupy, you can delete it

File: app/javascript/custom/dashboard.js [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+
+    function checkParent(t, elm) {
+        while (t.parentNode) {
+            if (t == elm) {
+                return true;
+            }
+            t = t.parentNode;
+        }
+        return false;
+    }
```
Comment (YilKanda) [inline app/javascript/custom/dashboard.js:164]: Check the indentation of your file (space: 2)

File: app/views/dashboard/_nav.html.erb [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+                <i class="fa-solid fa-chart-simple fa-fw mr-3"></i><span class="w-full inline-block pb-1 md:pb-0 text-sm">Estado de resultado</span>
+            </a>
+        </li>
+        <li class="my-2 md:my-0">
+            <a href="/dashboard/balance_sheet" class="block py-1 md:py-3 pl-1 align-middle text-gray-600 no-underline hover:text-indigo-400">
+                <i class="fas fa-balance-scale fa-fw mr-3"></i><span class="w-full inline-block pb-1 md:pb-0 text-sm">Balance general</span>
+            </a>
+        </li>
+    </ul>
+</div>
```
Comment (YilKanda) [inline app/views/dashboard/_nav.html.erb:40]: Check the indentation of your file (space: 2)

**Review by YilKanda — CHANGES_REQUESTED** [review]

File: app/assets/stylesheets/pages/dashboard.css [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+#sidebar span {
+  opacity: 0;
+  position: absolute;
+  transition: ease-in-out all .1s;
+}
+
+#sidebar:hover {
+  width: 290px;
+  box-shadow: 0 25px 50px -12px rgba(0, 0, 0, 0.25);
```
[the remark this answers is not in the document]
Comment (ever864) [reply app/assets/stylesheets/pages/dashboard.css:32]: Done

File: app/assets/stylesheets/pages/dashboard.css [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+  /*shadow-2xl*/
```
[the remark this answers is not in the document]
Comment (ever864) [reply app/assets/stylesheets/pages/dashboard.css]: Done

File: app/views/dashboard/_nav.html.erb [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+                </a>
+            </li>
+              <li class="my-2 md:my-0">
+                <a href="/dashboard/balance_sheet" class="block py-1 md:py-3 pl-1 align-middle text-gray-600 no-underline hover:text-indigo-400">
+                   <i class="fas fa-balance-scale fa-fw mr-3"></i><span class="w-full inline-block pb-1 md:pb-0 text-sm">Balance general</span>
+                </a>
+            </li>
+
+        </ul>
+    </div>
```
[the remark this answers is not in the document]
Comment (ever864) [reply app/views/dashboard/_nav.html.erb]: Done

File: app/views/dashboard/inventory.html.erb (hunk 1 above)
Comment (ever864) [reply app/views/dashboard/inventory.html.erb]: Done

File: app/views/dashboard/business_date.html.erb [hunk 10]
```diff
@@ -0,0 +1,10 @@
+<body class="h-screen bg-gray-100 font-sans">
+
+   <%= render 'nav' %>
+
+<div class="flex flex-row flex-wrap flex-1 flex-grow content-start pl-16">
+      <div id="dash-content" class="bg-gray-200 py-6 lg:py-0 w-full lg:max-w-sm flex flex-wrap content-start">
+        <h1 class="pl-2">Datos del negocio</h1>
+      </div>
+   </div>
+</body>
```
[the remark this answers is not in the document]
Comment (ever864) [reply app/views/dashboard/business_date.html.erb]: Done!

File: app/views/dashboard/providers.html.erb (hunk 2 above)
Comment (ever864) [reply app/views/dashboard/providers.html.erb]: Done

File: app/views/dashboard/sales_and_purcharses.html.erb (hunk 3 above)
Comment (ever864) [reply app/views/dashboard/sales_and_purcharses.html.erb]: Done

File: app/views/dashboard/statement_of_income.html.erb [hunk 11]
```diff
@@ -0,0 +1,8 @@
+<body class="h-screen bg-gray-100 font-sans">
+   <%= render 'nav' %>
+    <div class="flex flex-row flex-wrap flex-1 flex-grow content-start pl-16">
+      <div id="dash-content" class="bg-gray-200 py-6 lg:py-0 w-full lg:max-w-sm flex flex-wrap content-start">
+        <h1 class="pl-2">balance general</h1>
+      </div>
+    </div>
```
[the remark this answers is not in the document]
Comment (ever864) [reply app/views/dashboard/statement_of_income.html.erb]: Done

File: app/helpers/dashboard_helper.rb (hunk 4 above)
Comment (ever864) [reply app/helpers/dashboard_helper.rb]: Done

File: app/javascript/custom/dashboard.js (hunk 5 above)
Comment (ever864) [reply app/javascript/custom/dashboard.js]: Done!

File: app/views/dashboard/_nav.html.erb (hunk 6 above)
Comment (ever864) [reply app/views/dashboard/_nav.html.erb]: Done

File: app/views/dashboard/index.html.erb [hunk 12]
```diff
@@ -0,0 +1,396 @@
...
+
+    function checkParent(t, elm) {
+      while (t.parentNode) {
+        if (t == elm) {
+          return true;
+        }
+        t = t.parentNode;
+      }
+      return false;
+    }
```
Comment (YilKanda) [inline app/views/dashboard/index.html.erb:392]: This JS code, put it in an external file.

**Review by YilKanda — CHANGES_REQUESTED** [review]

File: app/views/dashboard/index.html.erb (hunk 12 above)
Comment (ever864) [reply app/views/dashboard/index.html.erb]: done

**Review by JuanequeX — DISMISSED** [review]
LGTM

**Review by YilKanda — APPROVED** [review]
LGTM!

**Review by acorzo — APPROVED** [review]
Gob job 👍

**Pull request merged**